### PR TITLE
Move api client initialization into api package

### PR DESCRIPTION
**Pull request by dron22**
This PR move the initialization of the api client into a function `NewClientFromParams` in `api` package. This removes duplicate code, as the api client was initialized in various commands (`heartbeat`, `today` and `todaygoal`). It is a preparation for implementation of sync offline command (https://github.com/wakatime/wakatime-cli/issues/404).

- Api client initialization is moved into `api.NewClientFromParams`. Main initialization logic is only moved and not modified.
- `legacyparams.API` and `legacyparams.Network` are unified into `api.Params`.
- Tests for api related parameter loading from viper is unified in `legacyparams` package and removed everywhere else. 
- `sync-offline-activity` command will be able to use the initialization logic via `api.NewClientFromParams` and does not need to implement it itself.

### CODE REVIEW

**Comment by alanhamlett** [comment]
Branch is outdated, needs a rebase.

**Comment by dron22** [comment]
> Branch is outdated, needs a rebase.

Thanks for the heads-up :+1:

**Review by gandarez — APPROVED** [review]
LGTM regardless the name `NewClientFromParams()` we've been discussing on Slack.

File: cmd/legacy/legacyparams/params.go [hunk 1]
```diff
@@ -32,7 +32,33 @@ var (
...
+	ProxyURL         string
+	SSLCertFilepath  string
+	Timeout          time.Duration
+	URL              string
+}
+
+// String implements fmt.Stringer interface.
+func (p API) String() string {
+	return fmt.Sprintf(
+		"api key: '%s', api url: '%s', plugin: '%s', timeout: %s, disable ssl verify: %t,"+
```
Comment (gandarez) [inline cmd/legacy/legacyparams/params.go:52]: Isn't it better to surround strings with double quote instead single quote?

Comment (dron22) [reply cmd/legacy/legacyparams/params.go:52]: Valid question. Actually using `%q` placeholder would be simpler, yes. But the reason I put them is that in the json log output, the message itself will be surrounded with double quotes as well. Consequently we would end up with escaped double quotes, which is slightly harder to read.
Example:
```
{"message":"api key: \"00635xxx\", api url: \"https://wakatime.com\", plugin: \"WakaTime/1.23\""}
```
versus:
```
{"message":"api key: '00635xxx', api url: 'https://wakatime.com', plugin: 'WakaTime/1.23'"}
```

**Review by gandarez — APPROVED** [review]

**Pull request merged**